GEECS-PythonAPI/geecs_python_api/analysis/scans/scan_data.py

```diff
@@ -119,6 +108,56 @@ def build_device_shot_path(tag: ScanTag, device_name: str, shot_number: int, fil
         file = scan_path / f'{device_name}' / f'Scan{tag[3]:03d}_{device_name}_{shot_number:03d}.{file_extension}'
         return file
 
+    @staticmethod
+    def get_latest_scan_tag(experiment: str, year: Optional[int] = None,
+                            month: Optional[int] = None, day: Optional[int] = None) -> 'ScanTag':
+        """
+        Locates the last generated scan of the given day.  If no day is given or info incomplete, then assume it's today
+        :param experiment: The name of the experiment
+        :param year: Optional, the year as a 4-digit int
+        :param month: Optional, the month as a 1/2-digit int
+        :param day: Optional, the day as a 1/2-digit int
+        :return: The ScanTag tuple representing the last, existing scan folder
+        """
+        if year is None or month is None or day is None:
+            today = datetime.today()
+            year = today.year
+            month = today.month
+            day = today.day
+
+        i = 1
+        new_scan_flag = True
+        while new_scan_flag:
```

Comment:
Good call, while it would be nice for a new day to start with the fresh scan 001 as the next, a previous day should say something if the latest scan for that day was to be scan000.  Probably return None is fine